encryption/src/main/java/org/apache/solr/encryption/EncryptionRequestHandler.java

```diff
@@ -615,4 +691,30 @@ private static class PendingKeyId {
       this.keyId = keyId;
     }
   }
+
+  /**
+   * Internal encryption request status.
+   */
+  private static class RequestStatus {
```

Comment:
could be an enum... I guess it's fine as is.

Reply:
The constructor is used line 557.